### PR TITLE
[pickers] Remove `components` and `componentsProps` props

### LINKED ISSUE #10335: [pickers] Remove deprecated `components` and `componentsProps` props
Following the conventions being followed across MUI products, MUI X deprecated `components` and `componentsProps` to replace `slots` and `slotProps`. 

In v7, these deprecated props should be removed. Here are the action items needed to be taken for completing this breaking change:
- [ ] Remove the capital 1st letter of all the slots
- [ ] Stop using `UncapitalizeObjectKeys` for all the `slots` props
- [ ] Remove the `components` and `componentsProps` on all pickers
- [ ] Make codemod for `components -> slots` conversion part of `preset-safe`.

### PR DESCRIPTION
**Pull request by alexfauquette**
Fix #10335

This PR has been done as follows:

1. Remove all occurrences of `UncapitalizeObjectKeys` in the code, and delete the file `/internals/utils/slots-migration.ts` that contains helpers to have `slots` and `components` together. This includes:
	 - Remove `UncapitalizeObjectKeys` every where in the code
	 - Remove `components`/`componentsProps` props definition when they are close.
	 - Replace `SlotsAndSlotProps` by explicit definition of slots
2. Fix the TS errors
	- Allows to remove all occurrences of `components`/`componentsProps` since most of their definition got removed before. The remaining ones get detected because subcomponents do not accept them now.
	- ALlws to lowercase all the `slots` attributes, because it leads to TS errors when using `slots?.Toolbar`.
3. Run scripts

Commits 1 and 2 needs a review, even if it's a boring one since most of the file got the same modification

### CODE REVIEW

**Comment by flaviendelangle** [comment]
We can't merge this one before the alpha branch is ready

File: packages/x-date-pickers/src/DatePicker/shared.tsx [hunk 1]
```diff
@@ -103,7 +90,7 @@ export function useDatePickerDefaultizedProps<TDate, Props extends BaseDatePicke
     };
   }, [themeProps.localeText]);
 
-  const slots = themeProps.slots ?? uncapitalizeObjectKeys(themeProps.components);
+  const slots = themeProps.slots;
```
Comment (LukasTy) [inline packages/x-date-pickers/src/DatePicker/shared.tsx:93]: Nitpick: It probably no longer makes sense to have a separate variable for this. 🤔 
```suggestion
```

File: packages/x-date-pickers/src/DatePicker/shared.tsx [hunk 2]
```diff
@@ -118,6 +105,5 @@ export function useDatePickerDefaultizedProps<TDate, Props extends BaseDatePicke
     minDate: applyDefaultDate(utils, themeProps.minDate, defaultDates.minDate),
     maxDate: applyDefaultDate(utils, themeProps.maxDate, defaultDates.maxDate),
     slots: { toolbar: DatePickerToolbar, ...slots },
```
Comment (LukasTy) [inline packages/x-date-pickers/src/DatePicker/shared.tsx:107]:
```suggestion
    slots: { toolbar: DatePickerToolbar, ...themeProps.slots },
```

**Review by LukasTy — APPROVED** [review]
Finally, the "great cleanup"! 💯 🚀 👍 

P.S. It might make sense to add a breaking change entry so that it would be picked up in the changelog generation. 🤔

**Pull request merged**